dist/origin-web-common-services.js

```diff
@@ -554,6 +554,70 @@ angular.module('openshiftCommonServices')
 });
 ;'use strict';
 
+angular.module("openshiftCommonServices").
+service("ApplicationsService", function($filter, $q, DataService) {
+
+  var getApplications = function(context) {
+    var hasDeploymentFilter = $filter('hasDeployment');
+    var hasDeploymentConfigFilter = $filter('hasDeploymentConfig');
+    var deploymentConfigs;
+    var deployments;
+    var replicationControllers;
+    var replicaSets;
+    var statefulSets;
+    var deferred = $q.defer();
+
+    var resolveApplications = function() {
+      // Don't resolve until we have all the types
+      if (deploymentConfigs && deployments && replicationControllers && replicaSets && statefulSets) {
+        var apiObjects = deploymentConfigs.concat(deployments)
+          .concat(replicationControllers)
+          .concat(replicaSets)
+          .concat(statefulSets);
+        deferred.resolve(_.sortBy(apiObjects, ['metadata.name', 'kind']));
```

Comment:
Actually this promise will never resolve if any of these fail, correct?

Reply:
Correct. This is how all the existing code worked but should be fixed.